src/base/bittorrent/torrenthandle.cpp

```diff
@@ -540,10 +540,15 @@ bool TorrentHandle::connectPeer(const PeerAddress &peerAddress)
     return true;
 }
 
-bool TorrentHandle::needSaveResumeData() const
+bool TorrentHandle::needSaveResumeData(bool saveResumeDataForCompletedTorrents) const
 {
+    // If a mutation has happened, then we definitely need to save out the resume data.
     if (m_needSaveResumeData) return true;
 
+    // (Perf) Otherwise, if we're complete, and the user doesn't want to write out this
+    // data, then don't do it.
+    if (!saveResumeDataForCompletedTorrents && this->isCompleted()) return false;
+
     return m_nativeHandle.need_save_resume_data();
```

Comment:
The problem here is due to using libtorrent RC_1_0.  That has an issue (fixed in RC_1_1) where this would return true every 15 minutes:

https://github.com/arvidn/libtorrent/blob/1d15e6bfb4a57f7cf07e62668b468580815060e0/ChangeLog
"need_save_resume_data() will no longer return true every 15 minutes"

Reply:
https://github.com/arvidn/libtorrent/issues/1637

Reply:
https://github.com/arvidn/libtorrent/pull/1942